File: LudoGameAndroid/app/src/main/java/com/ludogame/app/models/GamePlayer.kt
package com.ludogame.app.models

data class GamePlayer(
    val id: PlayerColor,
    val tokens: List<GameToken>,
    val isActive: Boolean = true
) {
    constructor(color: PlayerColor, isActive: Boolean = true) : this(
        id = color,
        tokens = (0 until 4).map { GameToken(id = it, owner = color) },
        isActive = isActive
    )

    val finishedTokenCount: Int
        get() = tokens.count { it.isFinished }

    val hasWon: Boolean
        get() = finishedTokenCount == 4
}

File: LudoGame/LudoGame/Models/GamePlayer.swift
import Foundation

struct GamePlayer: Identifiable, Equatable, Codable {
    let id: PlayerColor
    var tokens: [GameToken]
    var isActive: Bool

    init(color: PlayerColor, isActive: Bool = true) {
        self.id = color
        self.isActive = isActive
        self.tokens = (0..<4).map { GameToken(id: $0, owner: color) }
    }

    var finishedTokenCount: Int {
        tokens.filter(\.isFinished).count
    }

    var hasWon: Bool {
        finishedTokenCount == 4
    }
}
